compose/src/main/java/com/bottlerocketstudios/compose/map/GoogleMapsView.kt

```diff
@@ -49,101 +49,141 @@ fun GoogleMapsView(googleMapScreenState: GoogleMapScreenState, toolbarEnabled: B
     }
-
     val dialogVisibility = remember {
         mutableStateOf(value = false)
     }
 
+    val fullScreenMaps = remember(googleMapScreenState.businessList) {
+        derivedStateOf { googleMapScreenState.businessList.value.isEmpty() }
+    }
+
+    val showRetryButton = remember {
+        mutableStateOf(value = false)
+    }
+
     dialogVisibility.value = googleMapScreenState.yelpError.value != UserFacingError.NoError
 
     Column(
         modifier = Modifier
             .fillMaxSize()
             .background(Color.DarkGray)
     ) {
+        if (showRetryButton.value) {
+            RetryButton(retry = {
+                showRetryButton.value = false
+                googleMapScreenState.retrySearch()
+            })
+        }
         GoogleMap(
             properties = mapProperties,
-            uiSettings = mapUiSettings,
-            modifier = if (dialogVisibility.value) Modifier.fillMaxSize() else Modifier
+            modifier = if (fullScreenMaps.value) Modifier.fillMaxSize() else Modifier
                 .height(400.dp)
                 .fillMaxWidth(),
             cameraPositionState = googleCameraPositionState
-        )
+        ) {
+            if (googleMapScreenState.googleMarkers.value.isNotEmpty()) {
+                addMarkers(
+                    yelpMarkers = googleMapScreenState.googleMarkers.value,
+                    onclick = { marker ->
+                        Timber.d("${marker.id} ${marker.title}")
+                        googleMapScreenState.setSelectedMarker(marker.tag as YelpMarker)
+                        googleCameraPositionState.move(
+                            CameraUpdateFactory.newLatLng(
+                                LatLng(
+                                    marker.position.latitude,
+                                    marker.position.longitude
+                                )
+                            )
+                        )
 
-        AnimatedVisibility(dialogVisibility.value) {
-            when (val error = googleMapScreenState.yelpError.value) {
-                is UserFacingError.ApiError -> CustomAlertDialog(
-                    modifier = Modifier,
-                    title = error.title,
-                    message = error.description,
-                    onDismiss = {
-                        googleMapScreenState.resetError()
-                    }
+                        true
+                    },
+                    googleMapScreenState.yelpMarkerSelected.value
                 )
-                is UserFacingError.GeneralError -> CustomAlertDialog(
-                    modifier = Modifier,
-                    title = error.title,
-                    message = error.description,
-                    onDismiss = {
-                        googleMapScreenState.resetError()
-                    }
+            }
+
+            if (googleCameraPositionState.isMoving) {
+                val search = YelpLatLngSearch(
```

Comment:
Since we're using YelpLatLngSearch for a general purpose LatLng container.  We may want to rename it to just LatLng.  Also, if there is a standard class that already exists for such a purpose, we could use that.